Marlin/src/gcode/config/M217.cpp

```diff
@@ -60,20 +60,18 @@ void M217_report(const bool eeprom=false) {
 void GcodeSuite::M217() {
 
   bool report = true;
+  #if ENABLED(SINGLENOZZLE)
+    if (parser.seenval('S')) { report = false; const float v = parser.value_linear_units(); sn_settings.swap_length = constrain(v, 0, 500); }
+    if (parser.seenval('P')) { report = false; const int16_t v = parser.value_linear_units(); sn_settings.prime_speed = constrain(v, 10, 5400); }
+    if (parser.seenval('R')) { report = false; const int16_t v = parser.value_linear_units(); sn_settings.retract_speed = constrain(v, 10, 5400); }
 
-  if (parser.seenval('S')) { report = false; const float v = parser.value_linear_units(); sn_settings.swap_length = constrain(v, 0, 500); }
-  if (parser.seenval('P')) { report = false; const int16_t v = parser.value_linear_units(); sn_settings.prime_speed = constrain(v, 10, 5400); }
-  if (parser.seenval('R')) { report = false; const int16_t v = parser.value_linear_units(); sn_settings.retract_speed = constrain(v, 10, 5400); }
-
-  #if ENABLED(SINGLENOZZLE_SWAP_PARK)
-    if (parser.seenval('X')) { report = false; sn_settings.change_point.x = parser.value_linear_units(); }
-    if (parser.seenval('Y')) { report = false; sn_settings.change_point.y = parser.value_linear_units(); }
+    #if ENABLED(SINGLENOZZLE_SWAP_PARK)
+      if (parser.seenval('X')) { report = false; sn_settings.change_point.x = parser.value_linear_units(); }
+      if (parser.seenval('Y')) { report = false; sn_settings.change_point.y = parser.value_linear_units(); }
+    #endif
   #endif
-
-  if (parser.seenval('Z')) { report = false; sn_settings.z_raise = parser.value_linear_units(); }
+  if (parser.seenval('Z')) { report = false; z_raise = parser.value_linear_units(); }
```

Comment:
Making the global `z_raise` always exist even on machines that have only a single extruder makes no sense.